tenable/ad/widget/api.py

```diff
@@ -0,0 +1,260 @@
+from tenable.base.endpoint import APIEndpoint
+
+
+class WidgetsAPI(APIEndpoint):
+    _path = 'dashboards'
+    _schema = WidgetSchema()
+
+    def list(self, dashboard_id: str) -> List[Dict]:
+        '''
+        Retrieve all widgets
+
+        Returns:
+            list:
+                The list of widget objects
+
+        Examples:
+
+            >>> tad.widgets.list()
+        '''
+        return self._get(f"{dashboard_id}/widgets")
+
+    def create(self,
+               dashboard_id: str,
+               pos_x: int,
+               pos_y: int,
+               width: int,
+               height: int,
+               title: str) -> List[Dict]:
+        '''
+        Create a new widget
+
+        Args:
+            dashboard_id:
+                The dashboard instance identifier.
+            pos_x:
+                x-axis position for widget.
+            pos_y:
+                y-axis position for widget.
+            width:
+                width of widget.
+            height:
+                height of widget.
+            title:
+                title for widget.
+
+
+        Returns:
+            list[dict]:
+                The created widget object.
+
+        Examples:
+
+            >>> tad.widgets.create(
+            ...     dashboard_id=1,
+            ...     pos_x=1,
+            ...     pos_y=1,
+            ...     width=2,
+            ...     height=2,
+            ...     title='ExampleWidget',
+            ...     )
+        '''
+        payload = self._schema.dump(self._schema.load({
+            'posX': pos_x,
+            'posY': pos_y,
+            'width': width,
+            'height': height,
+            'title': title
+        }))
+
+        return self._schema.load(self._post(f'{dashboard_id}/widgets', json=payload))
+
+    def details(self,
+                dashboard_id: str,
+                widget_id: int) -> Dict:
+        '''
+        Retrieves the details for a specific widget.
+
+        Args:
+            dashboard_id: The dashboard instance identifier.
+            widget_id: The widget instance identifier
+
+        Returns:
+            dict:
+                the widget object.
+
+        Examples:
+
+            >>> tad.widget.details(1)
+        '''
+        return self._schema.load(self._get(f"{dashboard_id}/widgets/{widget_id}"))
+
+    def update(self,
+               dashboard_id: str,
+               widget_id: int,
+               **kwargs) -> Dict:
+        '''
+        update an existing widget
+
+        Args:
+            dashboard_id:
+                The dashboard instance identifier.
+            widget_id:
+                The dashboard instance identifier.
+            pos_x:
+                x-axis position for widget.
+            pos_y:
+                y-axis position for widget.
+            width:
+                width of widget.
+            height:
+                height of widget.
+            title:
+                title for widget.
+
```

Comment:
superfluous carriage return

Reply:
This has been resolved in latest commit.